### PR TITLE
Content(star): Thorne zytkow object sprite

### PR DESCRIPTION
**Pull request by RisingLeaf**
**Content (Artwork / Missions / Jobs)**

## Summary
The thorne-zytkow object, here is a link to the wikipedia article:
https://en.wikipedia.org/wiki/Thorne%E2%80%93%C5%BBytkow_object

Why I want to add this?
Well, I was thinking what interesting objects we could add to ES and I browsed some time in the interent, as you do when gathering ideas and I ended up finding this, an extremely seldom constellation that is in my opinion extremely cool. Imagine this being introduced during some sort of mission line and you jump into some system and see something crazy, I like that moment, I think you like this moment too.

So now, why is this draft?
I think the sprite is not quite there yet, but I cant find the issue, so I am begging for some review, please help me!

Some screenshots:
Screenshot 2023-01-17 at 3 50 17 PM
Screenshot 2023-01-17 at 3 50 08 PM

Reference:
image

## Artwork Checklist
 - [ ] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [ ] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: {{insert PR link}}
 - [ ] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}

### CODE REVIEW

File: data/map systems.txt [hunk 1]
```diff
@@ -25622,7 +25622,7 @@ system Rutilicus
 	fleet "Small Militia" 6000
 	fleet "Human Miners" 3000
 	object
-		sprite star/g5
+		sprite "star/thorne-zytkow"
```
Comment (RisingLeaf) [inline data/map systems.txt:25625]:
```suggestion
		sprite star/g5
```

**Comment by Saugia** [comment]
I'll get to an in-depth review later on, but this is definitely something I think should be a one-of in the game. I feel that's the best practice for any very specific star or planet type considering we aren't actually built around a billion star systems.

**Comment by RisingLeaf** [comment]
> I'll get to an in-depth review later on, but this is definitely something I think should be a one-of in the game. I feel that's the best practice for any very specific star or planet type considering we aren't actually built around a billion star systems.

I agree

**Comment by RisingLeaf** [comment]
The effect of a surprising object also only works once.

**Comment by RisingLeaf** [comment]
added some red and made the giant less transparent, as suggested by saugia.

**Comment by Zireael07** [comment]
"Less even" makes sense with the other object inside, though

**Comment by RisingLeaf** [comment]
Screenshot 2023-01-18 at 4 32 01 PM
Screenshot 2023-01-18 at 4 31 50 PM

So I tried to implement your suggestions @1010todd, it uses an 8k surface texture for the surface now to give more detail. I am not bale to reproduce the exact surface of other vanilla stars, I just dont know how it is made. This is a surface texture of the sun, so I think it should be fine?

I made the jetstream to be more in a cone, but maybe it is to broad now?

I did not push the changes yet, but you can see everything in the screenshots.

**Comment by 1010todd** [comment]
iirc current star sprites are made in GIMP so imitating it in Blender might be difficult. My first guess with trying to imitate it with Blender shader nodes would be to use noise with a lot of details and a bit of distortion mixed with musgrave with high detail and low dimension and 1~1.5 lacunarity. That should give you a bunch of extra details that's hopefully closer. Also ES stars are not really realistic so real sun texture still looks different though I do like the contrast between the brighter area and redder area so you could keep that and put the noise and musgrave textures over it I think.

Also yes, the beam might be a bit too broad now and less intense as well as being more sparse. I imagine the current sparse detail should be overlaying on a more solid beam. Also probably because of that where it erupts from the surface of the red star now looks odd like 2D image on a plane clipping into a 3D surface.  @GefullteTaubenbrust2 Made the Neutron star sprite so maybe he could give some input on this. Otherwise the length of the beam looks good to me.

Also you're still missing this brighter layer around the star. (The area _not_ highlighted in blue). Gradient node with Spherical/Quadratic Spherical should be able to do this.
ring

**Comment by RisingLeaf** [comment]
Another try 😆 
hope it improved at some points:

Screenshot 2023-01-19 at 1 58 52 PM

Screenshot 2023-01-19 at 1 59 00 PM

pinging @1010todd

**Comment by beccabunny** [comment]
How about adding a blue haze in place of the red haze at the poles, where it mixes with the flares?

**Comment by RisingLeaf** [comment]
> How about adding a blue haze in place of the red haze at the poles, where it mixes with the flares?

thorne-zytkow+

So more like this?

**Comment by beccabunny** [comment]
I like it!

**Comment by 1010todd** [comment]
The beam/flare looks pretty good to me now. Not sure if there's any other changes to the surface though, larger ES stars have visible short thin streaks all over it so this one should also have some to blend in better with the current sprites (see screencap below to see the details). I haven't found a way to fully replicate this yet but high detail noise with distortion plugged into musgrave vector seems to get somewhat closer than nothing. Also it seems like you're using additive blending, most stars in game use alpha blending so this probably should be the same.

Also I think the bright layer should be more opaque and blend in with the surface more. It probably should be made on the same object as the red star itself. Here's in game screencap of it.

image

**Comment by RisingLeaf** [comment]
Another try 😆 

ok, changes this time: some noise musgrave mixture spread all over the star in form of small brighter spots.
The edge of the star is much sharper now.

thorne-zytkow+

**Comment by 1010todd** [comment]
I can see the texture change now, it'll probably need some more variation I think, a mix of small ones and slightly bigger ones with slight color variation (yellow, orange, etc.) and use some of it to break up the bright areas because it looks blurry. 

> Edit 2: the edge might need to be broader

Yeah, also less bright I think.

> Edit: with this image size i would rather stay with additive shading because of filesize. not because I care for how big the game is, but because of loading the image in game could create a delay.

The major concern here is the visual, this is a big star and you can clearly see the starfield behind it in the screenshot, it really should be alpha blending. It won't impact game loading time visibly unless you make 30 frame animation for it or something.

Else if you really want to keep additive blending for some other reason at least something should be done so we can't see the starfield behind the red star, it looks weird.

**Comment by Ferociousfeind** [comment]
The star being additive-blended with the starfield behind it (and hazes, and everything else) implies it's sparse, letting light through it. Which intuitively would make sense if there's a neutron star inside it, blasting its insides with the two jet cones, essentially giving it a huge outward pressure to reduce its density.

But if that's the case, I would appreciate the internal neutron star to _not_ be additive blended, by making it a second sprite which is alpha-blended and is only the neutron star (plus maybe its jet cones?) so that the neutron star isn't _also_ low-density and pillowing out due to internal forces.

**Comment by RisingLeaf** [comment]
> The star being additive-blended with the starfield behind it (and hazes, and everything else) implies it's sparse, letting light through it. Which intuitively would make sense if there's a neutron star inside it, blasting its insides with the two jet cones, essentially giving it a huge outward pressure to reduce its density.
> 
> But if that's the case, I would appreciate the internal neutron star to _not_ be additive blended, by making it a second sprite which is alpha-blended and is only the neutron star (plus maybe its jet cones?) so that the neutron star isn't _also_ low-density and pillowing out due to internal forces.

I will switch to normal blending, but thanks for the comment

**Comment by Saugia** [comment]
Can we get the most recent image?

**Comment by RisingLeaf** [comment]
> Can we get the most recent image?

There are currently some issues with the shading, that is why I will probably use the method ferocious suggested

**Comment by RisingLeaf** [comment]
Here come some pictures taken from my barb on a dangerous flyby of the thorne-zytkow:
Screenshot 2023-02-13 at 6 50 01 PM
Screenshot 2023-02-13 at 6 50 07 PM
Screenshot 2023-02-13 at 6 50 18 PM

Things to note, similar to the neutron star I use a "core" sprite that just make the background black.
Tried to improve the blending of the edge.

**Comment by Saugia** [comment]
Overall, this looks a lot better now. The only concern I have left is whether or not the jets should be a different color. If they're coming from the internal neutron star, would it make more sense they're more part of the light red/whitish colors of that first ring layer around the star, since they're passing through it?

@beccabunny @1010todd @GefullteTaubenbrust2 Thoughts?

**Comment by beccabunny** [comment]
> Overall, this looks a lot better now. The only concern I have left is whether or not the jets should be a different color. If they're coming from the internal neutron star, would it make more sense they're more part of the light red/whitish colors of that first ring layer around the star, since they're passing through it?
> 
> @beccabunny @1010todd @GefullteTaubenbrust2 Thoughts?

Imho while it may make more sense, it looks much cooler like this, so I'd go with the current version

**Comment by GefullteTaubenbrust2** [comment]
Idk I have a lot of issues with it tbh, so sorry if this is a bit harsh.
I‘d maybe have a little bit of a gradient on the jets to make it blend better. Part of the reason the jets look a bit off also is the color dropoff is simply linear. It should have a hueshift into the blue in the darker region. In addition, the corona thing also looks far too regular in my eyes. The gradient is very circular and not irregular enough. Also the blending on the edges is odd to me. It’s clearly alpha blended, which looks wrong because it’s not physically accurate. I know the in game stars use alpha blending too, but the smooth gradient there at least gives the illusion of additive blending, so here it just looks off.
To be honest, the only thing that I really don’t have any significant complaints about is the star itself. 
But yeah, I much preferred some of the earlier versions of it infact, because many of these issues were less apparent, like they used additive blending. And that goes for the jets too, I think the orange glow looked much more organic.

**Comment by 1010todd** [comment]
In addition to what Gefu commented about the eruptions are too frequent and even (or both). I you want example think I already made one with nodes to cancel out most of the eruptions in this star maker thing (Just some simple Voronoi -> Colorramp -> multiply thing):

[ES StarGen V1 Blen331.zip](https://github.com/endless-sky/endless-sky/files/10784422/ES.StarGen.V1.Blen331.zip)

**Pull request closed without merging**